dev/arg_values.h

```diff
@@ -83,7 +83,7 @@ namespace sqlite_orm {
                 if(this->index < int(this->container.size()) && this->index >= 0) {
                     return this->currentValue;
                 } else {
-                    throw std::system_error(orm_error_code::index_is_out_of_bounds);
+                    throw std::system_error{orm_error_code::index_is_out_of_bounds};
```

Comment:
I just like the uniform constructor syntax with braces, that's the only reason I changed it. I can revert if it's not your taste :)
Though how exactly is C++20 different from previous standards in this respect? Is it similar to the relaxed strong enum construction rules (P0138R2)?

Reply:
It is just a note. I am also a fan of the uniform c-tor syntax. In C++20 or C++23 (don't remember exactly) parentheses act just like curly braces in c-tor calling syntax

Reply:
Hmm, do you have a link to that change? I am surprised because the parentheses allow for conversions w/o warnings and even play a role with edge cases like string and vector construction.

Reply:
https://en.cppreference.com/w/cpp/language/aggregate_initialization#Designated_initializers attention to the overload number 5

Reply:
Thanks, interesting read. It does apply to aggregate initialization, which system_error is not. Also they write that narrowing conversions are applicable using parantheses: "Until C++11, narrowing conversions were permitted in aggregate initialization, but they are no longer allowed, except that, as of C++20, narrowing conversions are allowed when aggregate initialization uses parentheses rather than braces."